src/LinAlg/matrix.h

```diff
@@ -171,6 +171,8 @@ namespace utl //! General utility classes and functions.
           size_t off1 = 0, int inc1 = 1,
           size_t off2 = 0, int inc2 = 1) const
     {
+      if (this->empty() || v.empty())
+        return 0.0;
       return this->dot(&v.front(),v.size(),off1,inc1,off2,inc2);
```

Comment:
use `v.data()` instead of `&v.front()` and the early exit wouldn't/shouldn't be neccessary

Reply:
ok